sdmetrics/single_table/privacy/dcr_overfitting_protection.py

```diff
@@ -0,0 +1,204 @@
+"""DCR Overfitting Protection metrics."""
+
+import numpy as np
+
+from sdmetrics.goal import Goal
+from sdmetrics.single_table.base import SingleTableMetric
+from sdmetrics.single_table.privacy.dcr_utils import calculate_dcr
+from sdmetrics.utils import get_columns_from_metadata, get_type_from_column_meta
+
+
+class DCROverfittingProtection(SingleTableMetric):
+    """DCR Overfitting Protection metric."""
+
+    name = 'DCROverfittingProtection'
+    goal = Goal.IGNORE
+    min_value = 0.0
+    max_value = 1.0
+
+    @classmethod
+    def _validate_inputs(
+        cls,
+        real_training_data,
+        synthetic_data,
+        real_validation_data,
+        metadata,
+        num_rows_subsample,
+        num_iterations,
+    ):
+        if num_rows_subsample is not None:
+            if not isinstance(num_rows_subsample, int):
+                raise ValueError('num_rows_subsample must be an integer.')
+            if num_rows_subsample < 1:
+                raise ValueError(
+                    f'num_rows_subsample ({num_rows_subsample}) must be greater than 1.'
+                )
```

Comment:
You could combine these into one check with 
`if not isinstance(num_rows_subsample, int) or num_rows_subsample < 1:`

Then the error would just be 'num_rows_subsample must be an integer greater than 1.'